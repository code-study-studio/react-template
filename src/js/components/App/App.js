// @flow

import React, { Component } from 'react';
import { render } from 'react-dom';
import { Route, Switch } from 'react-router-dom';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';
import Home from '../Home/Home';
import Route404 from '../Route404/Route404';
import './App.scss';

class App extends Component<{}> {
  render() {
    return (
      <ErrorBoundary>
        <Switch>
          <Route exact path='/' component={Home} />
          <Route component={Route404} />
        </Switch>
      </ErrorBoundary>
    );
  }
}

export default App;
