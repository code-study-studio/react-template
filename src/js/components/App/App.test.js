import React from 'react';
import App from './App';

describe('App', () => {

  it('should be defined', () => {
    expect(App).toBeDefined();
  });

  it('should render correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

});
