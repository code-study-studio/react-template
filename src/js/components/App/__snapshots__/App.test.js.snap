// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<ErrorBoundary>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</ErrorBoundary>
`;
